sstables/sstable_set.hh

```diff
@@ -114,6 +177,42 @@ public:
     incremental_selector make_incremental_selector() const;
 };
 
+class sstable_set_version {
+    lw_shared_ptr<versioned_sstable_set_data> _versioned_data;
+    schema_ptr _schema;
+    sstable_set_version_reference _prev;
+    mutable std::unordered_set<sstable_set_version*> _next;
```

Comment:
This _next variable is a set of pointers to versions that were derived directly from this version(by copying or modifying). It's mutable because we're adding a new version to it when copying(so in copy constructor and copy assignment, when we're working on a const reference)